packages/cms-cli/commands/logs.js

```diff
@@ -13,27 +14,74 @@ const { logDebugInfo } = require('../lib/debugInfo');
 const { loadConfig, checkAndWarnGitInclusion } = require('@hubspot/cms-lib');
 const { logger } = require('@hubspot/cms-lib/logger');
 const { logApiErrorInstance } = require('@hubspot/cms-lib/errorHandlers');
-const { toFile, outputLogs } = require('@hubspot/cms-lib/lib/logs');
+const { outputLogs } = require('@hubspot/cms-lib/lib/logs');
 const { getFunctionByPath } = require('@hubspot/cms-lib/api/function');
 const {
   getFunctionLogs,
   getLatestFunctionLog,
 } = require('@hubspot/cms-lib/api/results');
+const { base64EncodeString } = require('@hubspot/cms-lib/lib/encoding');
 
 const COMMAND_NAME = 'logs';
+const TAIL_DELAY = 5000;
 
-function getLogs(program) {
+const makeSpinner = (functionPath, portalIdentifier) => {
+  return ora(
+    `Tailing logs for '${functionPath}' on portal '${portalIdentifier}'.\n`
+  );
+};
+
+const makeTailCall = (portalId, functionId) => {
+  return async (after, spinner) => {
+    const latestLog = await getFunctionLogs(portalId, functionId, { after });
+    if (latestLog.results.length) {
+      spinner.clear();
+      outputLogs(latestLog);
+    }
+
+    return latestLog.paging.next.after;
+  };
+};
+
+const tailLogs = async (portalId, functionId, functionPath, portalName) => {
+  const tailCall = makeTailCall(portalId, functionId);
+  const spinner = makeSpinner(functionPath, portalName || portalId);
+  let after;
+
+  spinner.start();
+
+  try {
+    const latestLog = await getLatestFunctionLog(portalId, functionId);
+    after = base64EncodeString(latestLog.id);
+  } catch (e) {
+    // A 404 means no latest log exists(never executed)
+    if (e.statusCode !== 404) {
+      logApiErrorInstance(e, {
+        functionPath,
+        portalId,
+      });
+    }
+  }
+
+  return new Promise(() => {
+    setInterval(async () => {
+      // eslint-disable-next-line require-atomic-updates
+      after = await tailCall(after, spinner);
+    }, TAIL_DELAY);
```

Comment:
`setInterval` and async code can lead to inconsistent interval periods, and in some cases calls can occur out of desired order.  You can use a recursive `setTimeout` to ensure that at least `TAIL_DELAY` ms have elapsed since last invoked.

```javascript
const h = async () => {
  after = await tailCall(after, spinner);
  setTimeout(h, TAIL_DELAY);
};
h();
```

Reply:
Also, this promise isn't resolving?  Instead of promises this could be a use case for generators so you can just `yield` the results as they come in.

Reply:
I updated the async pattern with your suggestion as it is a safer bet. I think this could be a good case for a generator, but I think we should get this out the door and I can come back to it later to implement this as a generator.